cowprotocol/cowswap
# [EthFlow]#1291 creating order state

# Summary

Closes #1291 

Wiring up initial happy path for EthFlow

1. When EthFlow tx is sent, order will be in the `Creating` state
2. The link will point to Etherscan until the tx is mined
3. Once the tx is mined and the order is present in the backend, the order status will change to `Open`
4. The link will point to the Explorer

**WARNING**: Not yet handling cases when the order expires or fails/is rejected
**WARNING**: Only available in Goerli for now

# To Test

1. Turn ethflow on with `localStorage.setItem('enableEthFlow', '1')`
2. Place an order selling ETH for any token
* You should be able to place it successfully
* It prompt for an onchain tx
* Once tx is submitted, you should see the link to Etherscan in the UI (rather than Explorer)
* The pop up should say something like: `Swap xxx ETH for ...` (not WETH)
* The activity modal should say `Creating...` while the first tx is being mined and the `view details` link should lead to Etherscan
* The activity modal should say `Open` after the tx is mined and the `view details` link should lead to Explorer
* The order validTo should be the one set in the order (10 min)
* The order sell token should be ETH (not WETH)

3. Turn the ethflow flag off, refresh the page and perform a regular swap
* The app should behave as usual
